### PR TITLE
Update PostgreSQL.PostgreSQL to 16.3.1

### PR DESCRIPTION
**Pull request by FaserF**
Checklist for Pull Requests
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

Current 16.2.1 version does not work in winget anymore. Therefore this PR should also fix the general installation issues. 

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/153712)

### CODE REVIEW

**Comment by FaserF** [comment]
image

It displays 16.3-1 as version. So this should be correct?

File: manifests/p/PostgreSQL/PostgreSQL/16.3.1/PostgreSQL.PostgreSQL.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: PostgreSQL.PostgreSQL
+PackageVersion: 16.3.1
+InstallerType: exe
+Scope: machine
+InstallerSwitches:
+  Silent: --mode unattended
+  SilentWithProgress: --mode unattended --unattendedmodeui minimal
+Installers:
```
Comment (stephengillie) [inline manifests/p/PostgreSQL/PostgreSQL/16.3.1/PostgreSQL.PostgreSQL.installer.yaml:11]:
```suggestion
AppsAndFeaturesEntries:
- DisplayVersion: 16.3-1
Installers:
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
Ah, this is exactly why `DisplayVersion` exists - for when the application and website say one version number, but a different version number is written to the Registry. This way, when matching manifests in the repo with the data in the Registry, the package manager will know to match the Registry version with the `DisplayVersion`, while displaying the `PackageVersion` for `winget list` and similar output. 

The `16.2.1` version should have this as well, and also should have its `PackageVersion` corrected from `16.2`. But that would have to be in a separate PR.

**Comment by stephengillie** [comment]
One issue that I'm being reminded of is that all versions of `PostgreSQL.PostgreSQL` will need a `DisplayVersion` added. A PR adding `DisplayVersion` shouldn't be merged unless all other versions have it. If this manifest is the only one in the `PackageIdentifier` with this value, then it might cause issues with the other manifests.

**Comment by FaserF** [comment]
@stephengillie I have now changed this package to display the version 16.3 like the older versions and "fixed" the other 16.X versions (see the other PRs referenced to this PR). The versions before did it like I did with just writing 15.2.2 even if it is 15.2-2
I would leave those untouched.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**